### PR TITLE
Spike out preload/3 hook.

### PR DESCRIPTION
**Pull request by alanpeabody**
@beerlington really this could be in JaResource instead, neither place is perfect but I thought having it close to the relationship definition made sense at some point. It might be helpful for people not using JaResource here as well.

If this idea is not totally crazy I will work on some basic tests.

### CODE REVIEW

**Comment by beerlington** [comment]
This looks good. I'm thinking we could abstract it a little more in JaResource, but for this library I think it makes perfect sense. Couple typos in the docs that I'll comment on.

File: lib/ja_serializer/serializer.ex [hunk 1]
```diff
@@ -145,6 +146,29 @@ defmodule JaSerializer.Serializer do
...
   defcallback links(map, Plug.Conn.t) :: map
 
+  @doc """
+  A special callback that can be used to preload related data.
+
+  Unlike the other callbacks, this callback is ONLY executed on the top level
+  data being serialized. Also unlike any other callback when serializing a list
+  of data (eg: from an index action) it recieves the entire list, not each
+  individual post. When serializing a single record (eg, show, create, update)
+  a single record is recieved.
```
Comment (hunterboerner) [inline lib/ja_serializer/serializer.ex]: s/recieved/received

File: lib/ja_serializer/serializer.ex [hunk 2]
```diff
@@ -145,6 +146,29 @@ defmodule JaSerializer.Serializer do
...
+
+  The primary usecase of the callback is to preload all the relationships you
```
Comment (hunterboerner) [inline lib/ja_serializer/serializer.ex]: s/usecase/use case

**Comment by alanpeabody** [comment]
Thanks @hunterboerner!

**Pull request merged**